lib/Common/DataStructures/SizePolicy.h

```diff
@@ -32,6 +32,13 @@ struct PowerOf2Policy
     inline static hash_t GetBucket(hash_t hashCode, uint size, int modFunctionIndex)
     {
         AssertMsg(Math::IsPow2(size) && size > 1, "Size is not a power of 2.");
+
+        // we often deal with keys that differ in higher bits only, so smudge the entropy down a little 
+        // we do not need a perfect avalanche here, since this is for a hashtable lookup
+        // the following is sufficient and reasonably cheap
+        hashCode ^= (uint)hashCode >> 15;
+        hashCode ^= (uint)hashCode >> 7;
```

Comment:
there was a recent hacker news link [here](https://probablydance.com/2018/06/16/fibonacci-hashing-the-optimization-that-the-world-forgot-or-a-better-alternative-to-integer-modulo/) on using a fibonacii hash to map hash codes into buckets, and from the article, it has some nice properties that give us an even distribution across the bucket size.  Could we apply here?

Reply:
Interesting. I will try it.

Reply:
lmk I'm curious.

Reply:
from article, I think it may not work as well on small # of buckets.

Reply:
The article says basically that for small number of buckets there are funnels for the source values that are multiples of Fibonacci numbers larger than 21. I think that is harmless. 
Since the original number of bits is larger than the desired 4 bits, there would be funnels anyways and these kind of repeats do not seem to be likely.

I tried running the repro code and some benchmarks with the Fibonacci reducer and it did not seem to impact anything overall. 

The 15+7 shift that I suggested originally does not seem any better for a 4-bit bucket and is probably slightly slower on common hardware with decent IMUL. I will keep the Fibonacci reducer here.